modyn/selector/internal/selector_strategies/downsampling_strategies/rho_loss_downsampling_strategy.py

```diff
@@ -154,24 +200,36 @@ def training_set_producer() -> Iterable[tuple[list[tuple[int, float]], dict[str,
 
         total_keys_in_trigger, *_ = AbstractSelectionStrategy.store_training_set(
             self.rho_pipeline_id,
-            next_trigger_id,
+            target_trigger_id,
             self._modyn_config,
             training_set_producer,
             selector_storage_backend.insertion_threads,
         )
         logger.info(
-            f"Stored {total_keys_in_trigger} keys in the holdout set for trigger {next_trigger_id} "
+            f"Stored {total_keys_in_trigger} keys in the holdout set for trigger {target_trigger_id} "
             f"in rho pipeline {self.rho_pipeline_id}"
         )
 
     @staticmethod
-    def _get_holdout_sampling_query(main_pipeline_id: int, trigger_id: int, target_size: int) -> Select:
-        return (
-            select(SelectorStateMetadata.sample_key)
-            .filter(
+    def _get_sampling_query(
+        main_pipeline_id: int, trigger_id: int, probability: float, selector_storage_backend: AbstractStorageBackend
+    ) -> Select:
+        assert isinstance(selector_storage_backend, DatabaseStorageBackend)
+
+        def _session_callback(session: Session) -> None:
+            session.query(SelectorStateMetadata).filter(
                 SelectorStateMetadata.pipeline_id == main_pipeline_id,
                 SelectorStateMetadata.seen_in_trigger_id == trigger_id,
-            )
-            .order_by(func.random())  # pylint: disable=E1102
-            .limit(target_size)
```

Comment:
What is your reasoning behind changing the sampling logic? The reason we do it this way is because we want to get exactly `target_size` items. In your case, you'll get a different number of rows each time you sample. Is that a problem?